devices/zemismart.js

```diff
@@ -11,8 +11,9 @@ const fzLocal = {
         cluster: 'manuSpecificTuya',
         type: ['commandGetData', 'commandSetDataResponse'],
         convert: (model, msg, publish, options, meta) => {
-            const button = msg.data.dp;
-            const actionValue = tuya.getDataValue(msg.data.datatype, msg.data.data);
+            const dpValue = msg.data.dpValues[0];
```

Comment:
This is definitely how new converters should be written. See [an example](https://github.com/Koenkk/zigbee2mqtt.io/blob/a675a0d53a77112aac8b43786b8e4237f24f06e6/docs/advanced/support-new-devices/02_support_new_tuya_devices.md#5-adding-your-first-data-point) in my proposed change to the "Support new Tuya devices" page.

However, for existing converters, this means to introduce larger (structural) changes:

1. Most converters return on the first matched value, i.e. a for loop would only make a difference if data would be gathered for all DP values and returned after the loop (as in the documentation example).
2. I have no idea what happens if we begin to look at additional DP values.

For 1., I am afraid to introduce breakage, as we don't have tests for converters. For 2. I am afraid to introduce breakage as I don't have any of those device. 

Additionally, we don't know if these changes would make any difference (if a device never sends multiple DP values, the best we can do is not to break things). And it is a lot of additional work, the current replacements are mostly "robotic".

To play safe here, I did not introduce loops in the converters. 

What we could do is to introduce warning logs if there are additional DP values indeed. We could have a `firstDpValue(msg, meta, converterName)` function. As an example:

```js
    tuya_water_leak: {
        cluster: 'manuSpecificTuya',
        type: 'commandDataReport',
        convert: (model, msg, publish, options, meta) => {
            const dpValue = tuya.firstDpValue(msg, meta, 'tuya_water_leak');
   ...
```

This could log a warning that DP values were ignored and suggest to use a for loop including a pointer to the "Support new Tuya devices" page". However, the call is somewhat awkward in order to pass the information needed for logging.

In conclusion, I see the following options:

1. Bite the bullet and convert everything to loops (but perhaps in smaller chunks, i.e. merge this PR and begin to change to loops in subsequent PRs)
2. Keep it as in the current PR
3. Introduce logs (which would only emit warnings if multiple DPs occur in practice)
4. ?

I think I prefer option 3.

@hacker-cb What do you think?
@Koenkk How are such migrations handled in the zigbee2mqtt project typically?

Reply:
3 seems the best indeed, usually I merge such pr directly after release to have the longest possible test exposure (a new z2m release is made every 1st of the month)

Reply:
@hacker-cb hacker-cb commented now
I agree, option 3 is better.
Also it will be nice to send anonymous reports about such situations to the statistics server.

Reply:
> Also it will be nice to send anonymous reports about such situations to the statistics server.

There is no statistics server (and I prefer to keep it such for privacy reasons). Better to log with a proper message on what to do

Reply:
I adapted the PR accordingly, log message looks like this:

`zigbee-herdsman-converters:tuya_water_leak: Additional DP #4 with data {"dp":4,"datatype":2,"data":[0,0,0,75]} will be ignored! Use a for loop in the fromZigbee converter (see https://www.zigbee2mqtt.io/advanced/support-new-devices/02_support_new_tuya_devices.html)`

Reply:
Looks good to me!